common/djangoapps/enrollment/api.py

```diff
@@ -225,7 +225,8 @@ def update_enrollment(user_id, course_id, mode=None, is_active=None):
         }
 
     """
-    _validate_course_mode(course_id, mode)
+    if mode is not None:
+        _validate_course_mode(course_id, mode)
```

Comment:
AFAICT this changes present behavior; before this change, when `mode` was omitted from the request, the API would raise an error; now it will default the enrollment to 'honor' mode.